code/modules/shuttle/special.dm

```diff
@@ -255,10 +255,14 @@
 	use_power = NO_POWER_USE
 	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF
 	speech_span = SPAN_ROBOT
+	///Cumulative amount of credits required to pass
 	var/threshold = 500
+	///Amount of credits paid so far, will let you pass once threshold is met
+	var/list/payees = list()
```

Comment:
This means people have to pay twice if they pay partiall, then use another scanner. You can fix this by making it a static list, like approved_passengers is